applications/object_detection_torch/main.cpp

```diff
@@ -239,6 +240,9 @@ bool parse_arguments(int argc, char **argv, std::string &config_name, std::strin
       break;
 
     switch (c) {
+    case 'c':
+      config_name = optarg;
```

Comment:
Note that under current behavior a config passed with `--config` will be overwritten if another config is also passed as a positional argument. Can you confirm whether that is intended? OK to move forward?

Reply:
Inside a Holohub container, I was able to launch the app using both of the following:
```
$ ./run launch object_detection_torch  --extra_args "--config /workspace/holohub/applications/object_detection_torch/object_detection_torch.yaml"
$ ./run launch object_detection_torch  --extra_args /workspace/holohub/applications/object_detection_torch/object_detection_torch.yaml
```

The positional argument is handled right below the switch block if the user passes in the YAML config file without a `--config` switch, as noted in the second command above.

Reply:
Right, I wanted to note that the following command is also now a valid equivalent of those two examples above:
```sh
./run launch object_detection_torch  --extra_args "/workspace/holohub/applications/object_detection_torch/object_detection_torch.yaml --config path/to/my/bad/config.yaml
```

I think it is OK to favor backwards compatibility as you've noted and move forward.

Reply:
In that case, the value passed in with `--config` will be overwritten. 
I will add a check to avoid that but unlikely people will use both.

Reply:
```
$ ./run launch object_detection_torch  --extra_args "--config /workspace/holohub/applications/object_detection_torch/object_detection_torch.yaml path_to_bad.yaml"

[command] export PYTHONPATH=${PYTHONPATH}:/opt/nvidia/holoscan/lib/cmake/holoscan/../../../python/lib:/workspace/holohub/build/object_detection_torch/python/lib:/workspace/holohub && export HOLOHUB_DATA_PATH=/workspace/holohub/data && export HOLOSCAN_INPUT_PATH=/opt/nvidia/holoscan/data
[command] cd /workspace/holohub/build/object_detection_torch/applications/object_detection_torch
[command] ./object_detection_torch --data /workspace/holohub/data/object_detection_torch --config /workspace/holohub/applications/object_detection_torch/object_detection_torch.yaml path_to_bad.yaml
[warning] [main.cpp:245] Configuration file already set using --config, ignoring 'path_to_bad.yaml'
```

Reply:
Thanks, the warning rather than override looks good.

Could you please remove autoformatting style updates so that only the handful of functional changes are included?

Reply:
Done, restore the unformatted code.

Reply:
@gigony @tbirdso 

> Is the --config parsing requirement documented in the [HAP specification](https://docs.nvidia.com/holoscan/sdk-user-guide/cli/hap.html#)? I see environment variables but not CLI requirements. Am I overlooking it, or could that be added in the future?

This is not documented as its internal behavior: the CLI always passes `--config <file-path>` to the application when launching it. @gigony and I discussed this while designing distributed applications and the CLI. 
I will update Holoscan [CLI packaging doc](https://docs.nvidia.com/holoscan/sdk-user-guide/holoscan_packager.html) to include that requirement.